File: compositePattern/leafEquipment.cpp
#include "./leafEquipment.h"

Motherboard::Motherboard()
{
	_name = "Motherboard";
}

int Motherboard::NetPrice()
{
	return 1000;
}

int Motherboard::Power()
{
	return 500;
}

Cpu::Cpu()
{
	_name = "Cpu";
}

int Cpu::NetPrice()
{
	return 1000;
}

int Cpu::Power()
{
	return 500;
}

UselessEquipment::UselessEquipment()
{
	_name = "UselessEquipment";
}

int UselessEquipment::NetPrice()
{
	return 1;
}

int UselessEquipment::Power()
{
	return 10;
}

File: compositePattern/csharp/LeafEquipment.cs
class Motherboard : Equipment
{
	public Motherboard()
	{
		_name = "Motherboard";
	}

	public override int NetPrice()
	{
		return 1000;
	}


	public override int Power()
	{
		return 500;
	}
}

class Cpu : Equipment
{
	public Cpu()
	{
		_name = "Cpu";
	}

	public override int NetPrice()
	{
		return 1000;
	}


	public override int Power()
	{
		return 500;
	}
}


class UselessEquipment : Equipment
{
	public UselessEquipment()
	{
		_name = "UselessEquipment";
	}

	public override int NetPrice()
	{
		return 1;
	}


	public override int Power()
	{
		return 10;
	}
}
